> Cycle button: 👍 Filter speed: Actually seems quite slow, particularly given the number of visible columns and data items. Not immediately obvious where the bottleneck is, but is probably worth digging in to a bit.
> 
> Unfortunately I can't test the plotting as retrieving selog data seems to be broken:
> 
> ```
> 127.0.0.1 - - [14/Feb/2022 14:29:18] "GET /getJournal/sandals/journal_20_3.xml HTTP/1.1" 200 -
> 127.0.0.1 - - [14/Feb/2022 14:29:20] "GET /getNexusFields/sandals/cycle_20_3/70099 HTTP/1.1" 200 -
> [2022-02-14 14:29:24,081] ERROR in app: Exception on /getNexusData/sandals/cycle0_3/70099/:raw_data_1:selog:Prop1 [GET]
> Traceback (most recent call last):
>   File "/usr/lib64/python3.6/site-packages/flask/app.py", line 2447, in wsgi_app
>     response = self.full_dispatch_request()
>   File "/usr/lib64/python3.6/site-packages/flask/app.py", line 1952, in full_dispatch_request
>     rv = self.handle_user_exception(e)
>   File "/usr/lib64/python3.6/site-packages/flask/app.py", line 1821, in handle_user_exception
>     reraise(exc_type, exc_value, tb)
>   File "/usr/lib64/python3.6/site-packages/flask/_compat.py", line 39, in reraise
>     raise value
>   File "/usr/lib64/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
>     rv = self.dispatch_request()
>   File "/usr/lib64/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
>     return self.view_functions[rule.endpoint](**req.view_args)
>   File "backend/isisInternal.py", line 42, in getNexusData
>     fieldData = nexusInteraction.fieldData(instrument, cycle, runs, fields)
>   File "/home/tris/src/jv2/backend/nexusInteraction.py", line 109, in fieldData
>     data.append(runData(nxsFile, fields, run))
>   File "/home/tris/src/jv2/backend/nexusInteraction.py", line 58, in runData
>     runData.append(runTimes(file))
>   File "/home/tris/src/jv2/backend/nexusInteraction.py", line 31, in runTimes
>     mainGroup = file['raw_data_1']
> TypeError: list indices must be integers or slices, not str
> ```

I've noticed the filtering issues and am working towards debugging that
I believe the plotting is working in the spectra pr

--- a/frontend/chartview.cpp
+++ b/frontend/chartview.cpp
@@ -213,7 +213,7 @@ void ChartView::mouseMoveEvent(QMouseEvent *event)
     }
     else
     {
-        if (hovered_ != "")
+        if (!hovered_.isEmpty())
             emit showCoordinates(chart()->mapToValue(event->pos()).x(), chart()->mapToValue(event->pos()).y(), hovered_);
         else
             emit clearCoordinates();

--- a/frontend/nexusInteraction.cpp
+++ b/frontend/nexusInteraction.cpp
@@ -67,7 +67,8 @@ void MainWindow::customMenuRequested(QPoint pos)
     runNos.chop(1);
 
     QString url_str = "http://127.0.0.1:5000/getNexusFields/";
-    QString cycle = cyclesMap_[ui_->cycleButton->text()].replace(0, 7, "cycle").replace(".xml", "");
+    QString cycle = cyclesMap_[ui_->cycleButton->text()];
+    cycle.replace(0, 7, "cycle").replace(".xml", "");
     url_str += instName_ + "/" + cycle + "/" + runNos;
 
     HttpRequestInput input(url_str);
@@ -177,7 +178,8 @@ void MainWindow::contextGraph()
     if (runNos.size() == 0)
         return;
     QString url_str = "http://127.0.0.1:5000/getNexusData/";
-    QString cycle = cyclesMap_[ui_->cycleButton->text()].replace(0, 7, "cycle").replace(".xml", "");
+    QString cycle = cyclesMap_[ui_->cycleButton->text()];
+    cycle.replace(0, 7, "cycle").replace(".xml", "");
     QString field = contextAction->data().toString().replace("/", ":");
     url_str += instName_ + "/" + cycle + "/" + runNos + "/" + field;
 

